asf_core_data/pipeline/preprocessing/preprocess_epc_data.py

```diff
@@ -187,28 +195,61 @@ def load_and_preprocess_epc_data(
     # Do not save/overwrite the preprocessed data when not loading entire GB dataset
     # in order to prevent confusion.
 
+    if str(save_data) == "S3":
+        save_data = None
+    else:
+        raw_data_path = data_batches.get_batch_path(
+            base_config.RAW_EPC_DATA_PATH, data_path, batch, check_folder="input"
+        )
+
+        # Raw EPC can be found in data path?
+        raw_epc_exists = (Path(data_path) / raw_data_path).is_file()
+
     if n_samples is not None:
 
         save_data = None
         logging.warning(
             "You're not loading all samples so the processed data will not be saved!"
         )
 
-    if save_data or (save_data is not None and not os.path.isabs(save_data)):
+    if (save_data or (save_data is not None and not os.path.isabs(save_data))) and (
+        save_data is not None
+    ):
 
         save_data = data_path / save_data
 
         if subset != "GB":
             logging.warning("Careful! You're not loading the complete GB dataset.")
 
-    epc_df = epc_data.load_raw_epc_data(
-        data_path=data_path,
-        rel_data_path=rel_data_path,
-        subset=subset,
-        batch=batch,
-        usecols=usecols,
-        n_samples=n_samples,
-    )
+    if subset != "GB" and n_samples is not None:
+        logging.warning(
+            "Nation subsets do not work well in combination with low n_samples. Set n_samples=None for best results."
```

Comment:
the code still runs.. it just warns the user. is this the intention?

Reply:
Yes, it will tag the output data respectively as well, so it's fine do it but thought it's better to give a warning.